### DISCUSSION TITLE
Cloudflare KV backend

### DISCUSSION THREAD

**Discussion by james-pre** [zen-fs/core#104] [Ideas]
Hi @atty303,

I was checking out some of the awesome projects people have been making with ZenFS, and came across what seems to be an implementation of a Cloudflare KV backend (https://github.com/atty303/pob-web/blob/9f108afe02a23b3cded5fd4d5401220a0d2f9480/packages/driver/src/js/fs.ts). Would you be interested in an official backend for it, maybe `@zenfs/cloudflare`? 

Additionally, it looks like you reimplement the `WebAccess` backend. If there are any issues you are experiencing with the backend, please let me know so I can fix them.

**Comment by atty303** [comment]
https://github.com/atty303/pob-web/blob/9f108afe02a23b3cded5fd4d5401220a0d2f9480/packages/web/functions/api/kv/%5B%5Bname%5D%5D.ts

That FS relies on its own Cloudflare Workers HTTP API implementation, which I think is uncomfortable to provide as an official implementation since it depends largely on the requirements of the app, such as whether or not authentication with the API is required and the design of the KV key.

I think there was something wrong with WebAccess, but I don't remember. I'll look into it and if I remember, I'll submit an Issue.

**Reply by james-pre** [reply]
Thanks. I definitely would be willing to support an official backend for this, if you're interested.
